ext/npm-extension.js

```diff
@@ -258,6 +258,11 @@ exports.addExtension = function(System){
 	System.locate = function(load){
 		var parsedModuleName = utils.moduleName.parse(load.name),
 			loader = this;
+
+		if (load.metadata) {
+			load.metadata.parsedModuleName = parsedModuleName;
```

Comment:
I'm pretty sure it should. search for `createLoad` I think. I think it creates the metadata object.  Could be the helpers stuff maybe.